src/pause_area_map.c

```diff
@@ -0,0 +1,517 @@
+        unk6F4[index2].unkC = 0;
+        unk6F4[index2].unkE = 0;
+    }
+
+    sub_08126DDC(areamap);
+    unkVersatile = areamap->unk6E0.unk6;
+    sub_081286F0(unkVersatile, areamap->unk48[unkVersatile]);
+    sub_081275F8(areamap);
+
+    check_unk6E0(&areamap->unk6E0);
+    (&areamap->unk6E0)->unk10 |= 0x0001;
+
+    for (index = 0; index < 4; index++) {
+        gUnk_0203ACC0[index].unkE &= ~(0x0100 | 0x0200);
+    }
+}
+
+static void sub_08127FCC(void) {
+    struct AreaMap* areamap;
+    areamap = TaskGetStructPtr(gCurTask);
+
+    if (gUnk_0203ACC0[0].unkE & 0x1000 || gUnk_0203ACC0[1].unkE & 0x1000 || gUnk_0203ACC0[2].unkE & 0x1000 ||
+        gUnk_0203ACC0[3].unkE & 0x1000) {
+        m4aSongNumStart(506);
+        sub_08124EC8();
+        gCurTask->main = sub_081287F4;
+    }
+    else {
+        sub_0803D2A8(0, 0xff);
+        sub_0803D280(0x80, 0x7f);
+        gCurTask->main = sub_0812824C;
+        sub_08127760(areamap);
+    }
+}
+
+void sub_08128074(struct AreaMap* areamap) {
+    u32 playerId, r5;
+
+    for (playerId = 0; playerId < 4; playerId++) {
+        sub_0803E558(playerId);
+    }
+
+    check_unk6E0(&areamap->unk6E0);
+
+    if (areamap->unk48[areamap->unk6E0.unk6] == 2 && (areamap->unk40 & 0x2f) > 0xf) {
+        u8 unk6 = areamap->unk6E0.unk6;
+        if (areamap->unk6E0.unk8 > gUnk_083611F1[unk6][0] * 8) {
+            sub_0815604C(&areamap->unk60[0].unk0);
+        }
+        if (areamap->unk6E0.unkC > gUnk_083611F1[unk6][1] * 8) {
+            sub_0815604C(&areamap->unk60[1].unk0);
+        }
+        if (areamap->unk6E0.unk8 < gUnk_083611F1[unk6][2] * 8) {
+            sub_0815604C(&areamap->unk60[2].unk0);
+        }
+        if (areamap->unk6E0.unkC < gUnk_083611F1[unk6][3] * 8) {
+            sub_0815604C(&areamap->unk60[3].unk0);
+        }
+    }
+
+    for (r5 = 0; r5 < 2; r5++) {
+        sub_08127834(&areamap->unk6F4[r5]);
+    }
+
+    sub_08126DDC(areamap);
+    sub_08127010(areamap);
+    areamap->unk40++;
+
+    areamap->unk5E--;
+    if (areamap->unk5E < 1) {
+#ifndef NONMATCHING
+        register s16 *old, *ptr asm("r1");
+        register u32 val asm("r4");
+
+        ptr = &areamap->unk5C;
+        ++*ptr;
+        old = ptr;
+        while (TRUE) {
+            ptr = old;
+            val = gUnk_08D61B20[*ptr];
+            if (val == 0xff) {
+                *ptr = 0;
+            }
+            else {
+                CpuCopy32(gUnk_08D612E4[val], (void*)0x06002400, sizeof(gUnk_08D612E4[0]));
+                ++*old;
+                areamap->unk5E = val = gUnk_08D61B20[*old];
+                break;
+            }
+        }
+#else
+        ++areamap->unk5C;
+        while (gUnk_08D61B20[areamap->unk5C] == 0xff)
+            areamap->unk5C = 0;
+        CpuCopy32(gUnk_08D612E4[gUnk_08D61B20[areamap->unk5C]], (void*)0x06002400, sizeof(gUnk_08D612E4[0]));
+        ++areamap->unk5C;
+        areamap->unk5E = gUnk_08D61B20[areamap->unk5C];
```

Comment:
Do you think this is functionally equivalent? I want to have your opinion on it.